### ISSUE TITLE
[@typescript-eslint/quotes] False Positive in enum

### ISSUE THREAD

**Issue by RyanHirsch** [typescript-eslint/typescript-eslint#1351]
When using backtick quote rule for enum, it improperly flags single quotes and `--fix` causes invalid code to be produced

```JSON
{
  "rules": {
    quotes: [`off`],
    '@typescript-eslint/quotes': [`error`, `backtick`],
  }
}
```

```TS
export enum Recur {
  ASAP = 8,
  DAILY = 3,
  JOURNEY = 9,
  MONTHLY = 1,
  'ONE-OFF' = 7,
  WEEKLY = 2,
  YEARLY = 0,
}
```

**Expected Result**
```TS
export enum Recur {
  ASAP = 8,
  DAILY = 3,
  JOURNEY = 9,
  MONTHLY = 1,
  'ONE-OFF' = 7,
  WEEKLY = 2,
  YEARLY = 0,
}
```

**Actual Result**

```TS
export enum Recur {
  ASAP = 8,
  DAILY = 3,
  JOURNEY = 9,
  MONTHLY = 1,
  `ONE-OFF` = 7,
  WEEKLY = 2,
  YEARLY = 0,
}
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.12.0` |
| `@typescript-eslint/parser`        | `2.12.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.7.2` |

**Comment by a-tarasyuk** [comment]
@bradzacher Could you mark it as a bug? I'll fix it today.
